### PR TITLE
Azure.Messaging.ServiceBus sample for subscription filters

### PR DESCRIPTION
**Pull request by spelluru**
## Description
We have a TopicFilters sample for the Track 1 package Microsoft.Azure.ServiceBus, but we don't have a similar one for Track 2 SDK. I have created this samples that creates the following entities and then sends and receives messages to test the filters on subscription.

    1. A topic named `TopicFilterSampleTopic`.
    2. Subscriptions to the above topic with the following settings. 

       | Subscription name |  Filter type |  Filter expression | Action | 
       | --------- | ----------- | ------------------ | ------ | 
       | AllOrders | True Rule filter | `1=1` | |
       | ColorBlueSize10Orders | SQL filter | `color = 'blue' AND quantity = 10` | | 
       | ColorRed | SQL filter | `color = 'red'` | `SET quantity = quantity / 2;REMOVE priority; SET sys.CorrelationId = 'low'`| 
       | ColorRed | Correlation filter | `"label": "red", "correlationId": "high"` | | 

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X ] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [ X] Title of the pull request is clear and informative.
- [X ] If applicable, the code is properly documented.
- [ X] The code builds without any errors.

### CODE REVIEW

**Comment by spelluru** [comment]
@EldertGrootenboer - Please review this new sample I created and merge when you get a chance. Thanks.

File: samples/DotNet/Azure.Messaging.ServiceBus/TopicSubscriptionFilters/CreateTopicAndSubscriptions/CreateTopicsAndSubscriptions.csproj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (SeanFeldman) [inline samples/DotNet/Azure.Messaging.ServiceBus/TopicSubscriptionFilters/CreateTopicAndSubscriptions/CreateTopicsAndSubscriptions.csproj:5]: Suggest switching to .NET6.
.NET 5 End of Life is 2022-05-08

Comment (spelluru) [reply samples/DotNet/Azure.Messaging.ServiceBus/TopicSubscriptionFilters/CreateTopicAndSubscriptions/CreateTopicsAndSubscriptions.csproj]: Thanks @SeanFeldman.

**Comment by spelluru** [comment]
@EldertGrootenboer

**Review by EldertGrootenboer — APPROVED** [review]

**Review by EldertGrootenboer — APPROVED** [review]
Looks good, let's get this merged.

**Pull request merged**